agent/grpc-external/services/dataplane/get_envoy_bootstrap_params.go

```diff
@@ -40,9 +51,85 @@ func (s *Server) GetEnvoyBootstrapParams(ctx context.Context, req *pbdataplane.G
 		return nil, status.Error(codes.Unauthenticated, err.Error())
 	}
 
+	if s.EnableV2 {
+		// Get the workload.
+		workloadId := &pbresource.ID{
+			Name: proxyID,
+			Tenancy: &pbresource.Tenancy{
+				Namespace: req.Namespace,
+				Partition: req.Partition,
+				PeerName:  "local",
+			},
+			Type: catalog.WorkloadType,
+		}
+		workloadRsp, err := s.ResourceAPIClient.Read(ctx, &pbresource.ReadRequest{
+			Id: workloadId,
+		})
+		if err != nil {
+			// This error should already include the gRPC status code and so we don't need to wrap it
+			// in status.Error.
+			return nil, err
+		}
+		var workload pbcatalog.Workload
+		err = workloadRsp.Resource.Data.UnmarshalTo(&workload)
+		if err != nil {
+			return nil, status.Error(codes.Internal, "failed to parse workload data")
+		}
+
+		// Only workloads that have an associated identity can ask for proxy bootstrap parameters.
+		if workload.Identity == "" {
+			return nil, status.Errorf(codes.InvalidArgument, "workload %q doesn't have identity associated with it", req.ProxyId)
+		}
+
+		// todo (ishustava): ACL enforcement ensuring there's identity:write permissions.
+
+		// Get all proxy configurations for this workload. Currently we're only looking
+		// for proxy configurations in the same tenancy as the workload.
+		// todo (ishustava): we need to support wildcard proxy configurations as well.
+
+		proxyCfgList, err := s.ResourceAPIClient.List(ctx, &pbresource.ListRequest{
+			Tenancy: workloadRsp.Resource.Id.GetTenancy(),
+			Type:    mesh.ProxyConfigurationType,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		// Collect and merge proxy configs.
+		// todo (ishustava): sorting and conflict resolution.
+		bootstrapCfg := &pbmesh.BootstrapConfig{}
+		dynamicCfg := &pbmesh.DynamicConfig{}
+		for _, cfgResource := range proxyCfgList.Resources {
+			var proxyCfg pbmesh.ProxyConfiguration
+			err = cfgResource.Data.UnmarshalTo(&proxyCfg)
+			if err != nil {
+				return nil, status.Errorf(codes.Internal, "failed to parse proxy configuration data: %q", cfgResource.Id.Name)
+			}
+			if isWorkloadSelected(req.ProxyId, proxyCfg.Workloads) {
+				proto.Merge(bootstrapCfg, proxyCfg.BootstrapConfig)
+				proto.Merge(dynamicCfg, proxyCfg.DynamicConfig)
+			}
+		}
+
+		accessLogs := makeAccessLogs(dynamicCfg.GetAccessLogs(), logger)
+
+		return &pbdataplane.GetEnvoyBootstrapParamsResponse{
+			ClusterName:     workload.Identity,
+			Partition:       workloadRsp.Resource.Id.Tenancy.Partition,
+			Namespace:       workloadRsp.Resource.Id.Tenancy.Namespace,
+			BootstrapConfig: bootstrapCfg,
+			Datacenter:      s.Datacenter,
+			NodeName:        workload.NodeName,
+			NodeId:          workloadRsp.Resource.Id.Name,
```

Comment:
If we're just going to fill in the workload ID, can we just deprecate the field?

Reply:
It looks like this field is also not used by consul-dp and so we can just remove it as well.